src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -2346,182 +2354,37 @@ int xcldev::device::testM2m()
-    }
-    uint64_t boh_address = prop.paddr;
-
-    info->exec_bo_handle = xclAllocBO(handle, info->exec_size, 0, XCL_BO_FLAGS_EXECBUF);
-    info->exec_bo_ptr = reinterpret_cast<ert_start_kernel_cmd *>(xclMapBO(handle, info->exec_bo_handle, true));
-    if (info->exec_bo_ptr == nullptr) {
-        iops_free_unmap_bo(handle, info->exec_bo_handle, info->exec_bo_ptr, info->exec_size);
-        throw std::runtime_error("Cannot obtain exec buf BO");
-    }
-    std::memset(info->exec_bo_ptr, 0, info->exec_size);
-
-    //construct the exec buffer cmd to start the kernel.
-    int rsz = 19; // regmap array size
-    info->exec_bo_ptr->state = 0;
-    info->exec_bo_ptr->opcode = ERT_START_CU;
-    info->exec_bo_ptr->count = rsz;
-    info->exec_bo_ptr->cu_mask = (0x1 << 0);
-    info->exec_bo_ptr->data[rsz - 3] = boh_address;
-    info->exec_bo_ptr->data[rsz - 2] = (boh_address >> 32);
-}
 
-static void
-execute_cmds(xclDeviceHandle handle, std::vector<std::shared_ptr<exec_struct>>& cmds,
-    double& duration)
-{
-    auto start = std::chrono::high_resolution_clock::now();
-    for (auto& c : cmds) {
-        if(xclExecBuf(handle, c->exec_bo_handle))
-            throw std::runtime_error("Unable to issue exec buf");
-    }
-    for (auto& c : cmds) {
-        // c->wait();
-        while (c->exec_bo_ptr->state < ERT_CMD_STATE_COMPLETED) {
-            while (xclExecWait(handle, -1) == 0);
-        }
-        if(c->exec_bo_ptr->state != ERT_CMD_STATE_COMPLETED)
-            throw std::runtime_error("CU execution failed");
-    }
-    auto end = std::chrono::high_resolution_clock::now();
-    duration = (std::chrono::duration_cast<std::chrono::microseconds>
-    (end - start)).count();
-}
-
-static void
-run_iops_test(xclDeviceHandle handle, std::vector<std::shared_ptr<exec_struct>>& cmds,
-    std::vector<double>& iops_list, int first_used_mem, unsigned int cmd_per_batch)
-{
-    double duration = 0;
-    double total = 0;
-    for (unsigned int i = 0; i < cmd_per_batch; i++) {
-        exec_struct info = {0};
-        iops_alloc_init_bo(handle, first_used_mem, &info);
-        auto p = std::make_shared<exec_struct>(info);
-        cmds.push_back(p);
-    }
-
-    // Execute the cmds 5 times and get average duration
-    for (int i = 0; i < 5; i++) {
-        execute_cmds(handle, cmds, duration);
-        total += duration;
-    }
-    duration = total / 5.0;
-
-    //verify
-    for(auto& c : cmds) {
-        if(xclSyncBO(handle, c->out_bo_handle, XCL_BO_SYNC_BO_FROM_DEVICE, c->out_size, 0))
-            throw std::runtime_error("Cannot sync output BO from device");
-        if (strncmp(c->output_bo_ptr, "Hello World", 11))
-            throw std::runtime_error("Bad output result after CU execution");
-    }
-    iops_list.push_back(cmds.size() * 1000 * 1000 / duration);
-    std::cout << "Commands: " << cmds.size() << " iops: " << (cmds.size() * 1000 * 1000 / duration) << std::endl;
+    if (output.find("Overall Commands") == std::string::npos) {
```

Comment:
Is "overall commands" analogous to "passed"/ "failed"?

Reply:
Yes.

Reply:
For xbutil2, we are relying on exit codes instead of stdout. I see that you have return codes in the application so the porting should be fine.